include/etcpal/cpp/inet.h

```diff
@@ -0,0 +1,658 @@
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////
+// Operators
+////////////////////////////////////////////////////////////////////////////////////////////////////
+
+/// \addtogroup etcpal_cpp_inet
+/// @{
+
+// Special operators for comparing with the underlying types
+
+inline bool operator==(const EtcPalIpAddr& c_ip, const IpAddr& ip) noexcept
+{
+  return c_ip == ip.get();
+}
+
+inline bool operator!=(const EtcPalIpAddr& c_ip, const IpAddr& ip) noexcept
+{
+  return !(c_ip == ip);
+}
+
+inline bool operator==(const IpAddr& ip, const EtcPalIpAddr& c_ip) noexcept
+{
+  return ip.get() == c_ip;
+}
+
+inline bool operator!=(const IpAddr& ip, const EtcPalIpAddr& c_ip) noexcept
+{
+  return !(ip == c_ip);
+}
+
+inline bool operator==(const EtcPalSockAddr& c_ip, const SockAddr& ip) noexcept
+{
+  return c_ip == ip.get();
+}
+
+inline bool operator!=(const EtcPalSockAddr& c_ip, const SockAddr& ip) noexcept
+{
+  return !(c_ip == ip);
+}
+
+inline bool operator==(const SockAddr& ip, const EtcPalSockAddr& c_ip) noexcept
+{
+  return ip.get() == c_ip;
+}
+
+inline bool operator!=(const SockAddr& ip, const EtcPalSockAddr& c_ip) noexcept
+{
+  return !(ip == c_ip);
+}
+
+inline bool operator==(const EtcPalMacAddr& c_mac, const MacAddr& mac) noexcept
+{
+  return c_mac == mac.get();
+}
+
+inline bool operator!=(const EtcPalMacAddr& c_mac, const MacAddr& mac) noexcept
+{
+  return !(c_mac == mac);
+}
+
+inline bool operator==(const MacAddr& mac, const EtcPalMacAddr& c_mac) noexcept
+{
+  return mac.get() == c_mac;
+}
+
+inline bool operator!=(const MacAddr& mac, const EtcPalMacAddr& c_mac) noexcept
+{
+  return !(mac == c_mac);
+}
+
+// Standard operators
+
+inline bool operator==(const IpAddr& a, const IpAddr& b) noexcept
+{
+  return a.get() == b.get();
+}
+
+inline bool operator!=(const IpAddr& a, const IpAddr& b) noexcept
+{
+  return !(a == b);
+}
+
+inline bool operator<(const IpAddr& a, const IpAddr& b) noexcept
+{
+  return a.get() < b.get();
+}
+
+inline bool operator>(const IpAddr& a, const IpAddr& b) noexcept
+{
+  return b < a;
+}
+
+inline bool operator<=(const IpAddr& a, const IpAddr& b) noexcept
+{
+  return !(b < a);
+}
+
+inline bool operator>=(const IpAddr& a, const IpAddr& b) noexcept
+{
+  return !(a < b);
+}
+
+inline bool operator==(const SockAddr& a, const SockAddr& b) noexcept
+{
+  return a.get() == b.get();
+}
+
+inline bool operator!=(const SockAddr& a, const SockAddr& b) noexcept
+{
+  return !(a == b);
+}
+
+inline bool operator<(const SockAddr& a, const SockAddr& b) noexcept
+{
+  return a.get() < b.get();
+}
+
+inline bool operator>(const SockAddr& a, const SockAddr& b) noexcept
+{
+  return b < a;
+}
+
+inline bool operator<=(const SockAddr& a, const SockAddr& b) noexcept
+{
+  return !(b < a);
+}
+
+inline bool operator>=(const SockAddr& a, const SockAddr& b) noexcept
+{
+  return !(a < b);
+}
+
+inline bool operator==(const MacAddr& a, const MacAddr& b) noexcept
```

Comment:
Man, I can't wait for the space ship operator...

Reply:
_same_